test/tangle-prune.test.js

```diff
@@ -0,0 +1,122 @@
+// SPDX-FileCopyrightText: 2022 Mix Irving
+//
+// SPDX-License-Identifier: LGPL-3.0-only
+
+const test = require('tape')
+const bipf = require('bipf')
+const Testbot = require('./helpers/testbot')
+const tanglePrune = require('../lib/tangle-prune')
+
+const chars = 'abcABC123=+? '.split('')
+const encodedLength = (obj) => bipf.encodingLength(obj)
+const randomChar = () => chars.sort(() => (Math.random() < 0.5 ? -1 : +1))[0]
+const randomText = (length) => {
+  let output = ''
+  while (output.length < length) output += randomChar()
+  return output
+}
+
+test('tangle prune', async (t) => {
+  const ssb = Testbot()
+  const ssbId = ssb.id
```

Comment:
:hot_pepper: (emoji for potential spicy problem)
With different feedFormats, the "id" of an feedId is of different lengths, this means that if `recps` is a GroupId + 15 feedIds.... they could be "sigil" style or "ssb-uri" style.

We need to clarify in a spec "when you are DM'ing a person to add them to a group, which feedId do you use?"
Is it their rootFeedId (which is bendy butt ssb-uri?)

cc @arj03

Reply:
EASY solution @Powersource - assume all ssb-uri of the longest length, which would currently be:


https://github.com/ssbc/ssb-uri2/blob/main/test/fixtures.js#L17-L26
this long:
```
ssb:feed/gabbygrove-v1/FY5OG311W4j_KPh8H9B2MZt4WSziy_p-ABkKERJdujQ=
```
or wait... what is this
```
ssb:feed/buttwoo-v1/FY5OG311W4j_KPh8H9B2MZt4WSziy_p-ABkKERJdujQ=/Z0rMVMDEO1Aj0uPl0_J2NlhFB2bbFLIHlty_YuqArFq=
```

Yeah we need a spec

Reply:
Can they be URI style though? Have we allowed that anywhere actually?

Reply:
Also worth considering this

https://github.com/ssbc/ssb-tribes2/blob/dc4e7bf6fbe0ececd780d7e33cdb95659d7b9577/lib/tangle-prune.js#L8-L15